tokio/src/sync/mutex.rs

```diff
@@ -105,21 +106,40 @@ pub struct Mutex<T> {
 /// A handle to a held `Mutex`.
 ///
 /// As long as you have this guard, you have exclusive access to the underlying `T`. The guard
-/// internally keeps a reference-couned pointer to the original `Mutex`, so even if the lock goes
-/// away, the guard remains valid.
+/// internally borrows the `Mutex`, so the mutex will not be dropped while a guard exists.
 ///
 /// The lock is automatically released whenever the guard is dropped, at which point `lock`
 /// will succeed yet again.
 pub struct MutexGuard<'a, T> {
     lock: &'a Mutex<T>,
 }
 
+/// An owned handle to a held `Mutex`.
+///
+/// This guard is only available from a `Mutex` that is wrapped in an [`Arc`]. It
+/// is identical to `MutexGuard`, except that rather than borrowing the `Mutex`,
+/// it clones the `Arc`, incrementing the reference count. This means that
+/// unlike `MutexGuard`, it will have the `'static` lifetime.
+///
+/// As long as you have this guard, you have exclusive access to the underlying `T`. The guard
+/// internally keeps a reference-couned pointer to the original `Mutex`, so even if the lock goes
+/// away, the guard remains valid.
+///
+/// The lock is automatically released whenever the guard is dropped, at which point `lock`
+/// will succeed yet again.
```

Comment:
Sure, will do. There was some existing inconsistent wrapping in this file so I wasn't sure what style to follow.

Reply:
The 80 characters ensure that the github view doesn't wrap lines in split mode, at least on my screen, so I like that width. It's also used relatively widely.

Reply:
You forgot this paragraph :sweat_smile:

Reply:
I definitely also prefer 80 characters. :) I just also try to avoid changing the wrapping of existing comments that I haven't actually changed, so the git blame is clearer.